source/slang/slang-ast-decl.cpp

```diff
@@ -118,4 +119,21 @@ bool isLocalVar(const Decl* decl)
     return false;
 }
 
+ThisTypeDecl* InterfaceDecl::getThisTypeDecl()
+{
+    for (auto member : members)
+    {
+        if (auto thisTypeDeclCandidate = as<ThisTypeDecl>(member))
+        {
+            return thisTypeDeclCandidate;
+        }
+    }
+    SLANG_UNREACHABLE("InterfaceDecl does not have a ThisType decl.");
+}
+
+InterfaceDecl* ThisTypeConstraintDecl::getInterfaceDecl()
+{
+    return as<InterfaceDecl>(parentDecl->parentDecl);
+}
+
```

Comment:
I'm happy to see this cleanup be put in place. I think this is a much more sound way for us to be modeling the `This` type inside of an `interface`, and I believe it sets the stage for us to be better able to handle proper polymorphism in the context of `class` types (or even `struct`s, if we want to allow them to `override` inherited method implementations).